scout/commands/delete/delete_command.py

```diff
@@ -8,6 +8,106 @@
 LOG = logging.getLogger(__name__)
 
 
+@click.command("variants", short_help="Delete variants for one or more cases")
+@click.option("-u", "--user", help="User running this command (email)", required=True)
+@click.option("-c", "--case-id", help="Case id")
+@click.option(
+    "-status",
+    type=click.Choice(["solved", "archived", "migrated", "active", "inactive", "prioritized"]),
+    multiple=True,
+    default=["solved", "archived", "migrated"],
+    help="Restrict to cases with specified status",
+)
+@click.option("-older-than", type=click.INT, default=0, help="Older than (months)")
+@click.option("-rank-threshold", type=click.INT, default=5, help="With rank threshold lower than")
+@click.option("-variants-threshold", type=click.INT, help="With more variants than")
+@click.option(
+    "--dry-run",
+    is_flag=True,
+    help="Perform a simulation without removing any variant",
+)
+@with_appcontext
+def variants(
+    user,
+    case_id,
+    status,
+    older_than,
+    rank_threshold,
+    variants_threshold,
+    dry_run,
+):
+    """Delete variants for one or more cases"""
+
+    user_obj = store.user(user)
+    if user_obj is None:
+        click.echo(f"Could not find a user with email '{user}' in database")
+        return
+
+    total_deleted = 0
+    items_name = "deleted variants"
+    if dry_run:
+        click.echo("--------------- DRY RUN COMMAND ---------------")
+        items_name = "estimated deleted variants"
+    else:
+        click.confirm("Variants are going to be deleted from database. Continue?", abort=True)
+
+    case_query = store.build_case_query(case_id, status, older_than)
+
+    # Estimate the average size of a variant document in database
+    avg_var_size = store.collection_stats("variant").get("avgObjSize", 0)  # in bytes
+
+    # Get all cases where case_query applies
+    n_cases = store.case_collection.count_documents(case_query)
+    cases = store.cases(query=case_query)
+    filters = (
+        f"Rank-score threshold:{rank_threshold}, case n. variants threshold:{variants_threshold}."
+    )
+    for nr, case in enumerate(cases, 1):
```

Comment:
Suggest to put the logic here in a function and let the cli only handle input and output. Makes easier to refactor into smaller functions

Reply:
Yes you'd have smaller functions but on the other hand it would become more hard to follow the programming logic, and to understand what the function does you'd have to open another file. When I wrote the delete_variants function it was obscenely complicated and the cyclomatic complexity was very high. I refactored and moved stuff in 3 new functions:

- collection_stats
- build_case_query
- remove_variants_event

Plus it's already calling `evaluated_variants`. 
Now the complexity is fine (codefactor agrees) and it's very easy to understand what is happening. I wouldn't fragment it more than this